### ISSUE TITLE
No way to unlock a file?

### ISSUE THREAD

**Issue by useronym** [ProjectBorealis/UE4GitPlugin#64]
![Screenshot 2022-11-03 154737](https://user-images.githubusercontent.com/4563753/199753239-ee10a757-165f-411e-ae07-0f09234fded8.png)

 1. The revert option only appears if the local file is modified
 2. Reverting a file does not unlock it

Obviously I can unlock through command line, but this seems like a fairly basic feature that is either missing or not working properly.

Plugin version: 3.08

**Comment by mastercoms** [comment]
What does your .gitattributes look like?

**Comment by mastercoms** [comment]
Ah never mind, I misunderstood what you were reporting. The revert or check in features should work fine to unlock, I will see why exactly it could not show up in your case, might be a UE5 change.

**Comment by useronym** [comment]
Thanks for the quick answer!

> What does your .gitattributes look like?

Just in case it's gonna be relevant, here you go:

```
# Unreal Engine file types.
*.uasset filter=lfs diff=lfs merge=lfs -text lockable
*.umap filter=lfs diff=lfs merge=lfs -text lockable

# Raw Content file types.
*.fbx filter=lfs diff=lfs merge=lfs -text lockable
*.3ds filter=lfs diff=lfs merge=lfs -text lockable
*.psd filter=lfs diff=lfs merge=lfs -text lockable
*.png filter=lfs diff=lfs merge=lfs -text lockable
*.mp3 filter=lfs diff=lfs merge=lfs -text lockable
*.wav filter=lfs diff=lfs merge=lfs -text lockable
*.xcf filter=lfs diff=lfs merge=lfs -text lockable
*.jpg filter=lfs diff=lfs merge=lfs -text lockable
*.mp4 filter=lfs diff=lfs merge=lfs -text lockable

# Anything in `/RawContent` dir.
/RawContent/**/* filter=lfs diff=lfs merge=lfs -text lockable
```

**Comment by mastercoms** [comment]
In any case, I think revert as unlock when unmodified is confusing and unclear to the user, even if it works in UE4. So I will see about adding another source control action for unlocking. If it's not possible, will try my best to debug the UE5 issue.

**Comment by useronym** [comment]
Yes, having "Unlock" or "Check-in" would make a lot more intuitive sense I think. I had to look at the gifs in the README to figure out that "Revert" is supposed to check in.

**Comment by mastercoms** [comment]
![image](https://user-images.githubusercontent.com/2672245/199856411-ba4b49c1-192c-4e97-bf7d-c03d58fdccb0.png)
Check in is supposed to check in, in your case it's not showing up properly. Check in submits the asset to the repo. Revert unlocks the file. In your case, it seems like the lock is just not being detected (which is why #65 happens as well).

**Comment by mastercoms** [comment]
![checked out](https://user-images.githubusercontent.com/2672245/199857279-a6fd4ac4-69e8-4bc8-b529-13c714b250f5.png)

I've tested this in a UE5 project as well, 5.0.3 launcher. It's working fine. What operating system are you running on? And sometimes, the git-lfs executables in the plugin root folder are not copied due to filters, do you have those in the plugin folder?

**Comment by useronym** [comment]
OS is Windows 10.
I have git-lfs.exe executable in plugin's root folder. But I also have the standard version of git-lfs on my path, not sure if that could be confusing the plugin into using the wrong binary.

I have found a way to consistently reproduce the issue on my end:
 1. Check-out a file (material in this case)
     - Checkmark on the file is red 
     - At this point, I can still check it back in
 2. Make modifications to the file and save
     - Checkmark on the file has turned yellow
     - At this point, I get the warning that file is not checked out
     - I can no longer check the file in



https://user-images.githubusercontent.com/4563753/199930732-333c12b0-4bed-4006-abb9-6bc1926af140.mp4

**Comment by mastercoms** [comment]
How is your username set in the source control settings? It says checked out by useronym, you might not have the same LFS username set.

**Comment by useronym** [comment]
Hmm well spotted. How do you set your lfs username? From some googling it seems like it's maybe the Git server you're using that provides the username? Using GitHub if that makes a difference. I have only configured
```
git config --global user.name "Adam Krupicka"
```
which is obviously different from whatever lfs lock is using 🤔

**Comment by mastercoms** [comment]
Currently, the plugin does not automatically determine this (although it could). You can set it here:

![settings menu](https://user-images.githubusercontent.com/2672245/199948850-866d2a21-a259-4edc-bec6-f8078f6b9661.png)

![settings window](https://user-images.githubusercontent.com/2672245/199948903-2347bbee-297f-404d-8123-10c006e063ce.png)

**Comment by useronym** [comment]
Hmm it is already set to my name

![Screenshot 2022-11-04 113801](https://user-images.githubusercontent.com/4563753/199953416-0fdf4e62-33ba-4e64-999c-c74c15ab9825.png)


But yes for some reason the username it's checked out under changes to `useronym` after saving the checked out asset.

**Comment by mastercoms** [comment]
I meant that you should be using useronym as the Git LFS name there. You have to match what the server identifies your lock name as.

**Comment by useronym** [comment]
Ah of course. That makes sense. Now everything seems to work.

Kinda my fault for not reading the documentation in README properly. In my defense, when I started scrolling down and saw a long section about "status branches" I decided to stop reading as it didn't seem relevant :P So maybe the docs should say more clearly how to setup your project to have things work.

Thanks a lot for the help with figuring out this issue!
